core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/NetworkConfigurationValidator.java

```diff
@@ -0,0 +1,31 @@
+package com.sequenceiq.cloudbreak.controller.validation;
+
+import java.util.Set;
+
+import org.apache.commons.net.util.SubnetUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.controller.BadRequestException;
+import com.sequenceiq.cloudbreak.domain.InstanceGroup;
+import com.sequenceiq.cloudbreak.domain.Network;
+
+@Component
+public class NetworkConfigurationValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkConfigurationValidator.class);
+
+    public void validateNetworkForStack(Network network, Set<InstanceGroup> instanceGroups) {
+        SubnetUtils utils = new SubnetUtils(network.getSubnetCIDR());
+        String[] addresses = utils.getInfo().getAllAddresses();
+        int nodeCount = 0;
+        for (InstanceGroup instanceGroup : instanceGroups) {
+            nodeCount += instanceGroup.getNodeCount();
+        }
+        if (addresses.length < nodeCount) {
+            LOGGER.error("Invalid nodecount because '{}' node can not assignable into '{}' addresses", nodeCount, addresses.length);
```

Comment:
please change the message to: Cannot assign more than {} addresses in the selected subnet.